rom/dev/src/flow/cold_reset/fw_processor.rs

```diff
@@ -102,52 +114,82 @@ impl FirmwareProcessor {
         })
     }
 
-    /// Download the image
+    /// Process mailbox commands
     ///
     /// # Arguments
     ///
-    /// * `env` - ROM Environment
+    /// * `soc_ifc` - SOC Interface
+    /// * `mbox` - Mailbox
+    /// * `pcr_bank` - PCR Bank
+    /// * `sha384` - SHA384
     ///
     /// # Returns
+    /// * `MailboxRecvTxn` - Mailbox Receive Transaction
     ///
     /// Mailbox transaction handle. This transaction is ManuallyDrop because we
     /// don't want the transaction to be completed with failure until after
     /// handle_fatal_error is called. This prevents a race condition where the SoC
     /// reads FW_ERROR_NON_FATAL immediately after the mailbox transaction
     /// fails, but before caliptra has set the FW_ERROR_NON_FATAL register.
-    fn download_image<'a>(
+    fn process_mailbox_commands<'a>(
         soc_ifc: &mut SocIfc,
         mbox: &'a mut Mailbox,
+        pcr_bank: &PcrBank,
+        sha384: &mut Sha384,
     ) -> CaliptraResult<ManuallyDrop<MailboxRecvTxn<'a>>> {
         soc_ifc.flow_status_set_ready_for_firmware();
 
-        cprint!("[afmc] Waiting for Image ");
+        cprint!("[afmc] Waiting for Measurement(s) or Image ");
+        let mut measurement_count = 0;
         loop {
             if let Some(txn) = mbox.peek_recv() {
-                if txn.cmd() != Self::MBOX_DOWNLOAD_FIRMWARE_CMD_ID {
-                    cprintln!("Invalid command 0x{:08x} received", txn.cmd());
-                    txn.start_txn().complete(false)?;
-                    continue;
+                match txn.cmd() {
+                    Self::MBOX_STASH_MEASUREMENT_CMD_ID => {
+                        if measurement_count == Self::MEASUREMENT_MAX_COUNT {
+                            cprintln!(
+                                "[afmc] Maximum supported number of measurements already received, ignoring."
+                            );
+                            txn.start_txn().complete(false)?;
+                            continue;
+                        }
+
+                        // Re-borrow mailbox to work around https://github.com/rust-lang/rust/issues/54663
+                        let txn = mbox
+                            .peek_recv()
+                            .ok_or(CaliptraError::FW_PROC_MAILBOX_STATE_INCONSISTENT)?;
+
+                        // Extend the measurement into PCR1.
```

Comment:
Why PCR1? This is for extending measurements of SoC firmware into DPE. I don't think they should impact the PCR containing Caliptra's own firmware measurements.

Can we pick a different PCR to separate them from Caliptra RT measurements?

Reply:
I also think we should also pass these measurements up in the FHT so RT can use them to populate the TcbInfos in the cert that DPE generates.

Reply:
> Why PCR1? This is for extending measurements of SoC firmware into DPE. I don't think they should impact the PCR containing Caliptra's own firmware measurements.
> 
> Can we pick a different PCR to separate them from Caliptra RT measurements?

This is now being extended to PCR31.

Reply:
> I also think we should also pass these measurements up in the FHT so RT can use them to populate the TcbInfos in the cert that DPE generates.

Can you file a separate issue for this?

Reply:
> This is now being extended to PCR31.

👍 

> Can you file a separate issue for this?

Actually, I think it should be fine for RT to parse these values from the PCR log.